### PR TITLE
Implement different file types for Entry.create_datum/Data.__new__ 

### PR DESCRIPTION
**Pull request by zngguvnf**
- wav
- flac
- ogg
- mat

### CODE REVIEW

File: jbof.py [hunk 1]
```diff
@@ -127,7 +138,15 @@ class Datum(numpy.ndarray):
     def __new__(cls, metafile):
         with metafile.open() as f:
             metadata = json.load(f)
-        data = numpy.load(metadata['_filename'])
+        extension = splitext(metadata['_filename'])[1]
```
Comment (bastibe) [inline jbof.py]: please use `pathlib.Path.suffix`

File: jbof.py [hunk 2]
```diff
@@ -105,15 +108,23 @@ def create_datum(self, name, data, metadata, fileformat='npy', samplerate=None):
...
+        Currently, only `fileformat`= ['msgpack', 'csv'] are not implemented.
         """
-        if not fileformat == 'npy':
-            raise NotImplementedError('Only npy is supported')
+        datafilename = self.directory / (name + '.' + fileformat)
+        if fileformat == 'npy':
+            numpy.save(datafilename, data)
+        elif fileformat in ['wav', 'flac', 'ogg']:
+            if samplerate:
+                soundfile.write(str(datafilename), data, int(samplerate))
```
Comment (bastibe) [inline jbof.py:120]: please add `samplerate` to metadata.

File: jbof.py [hunk 3]
```diff
@@ -18,6 +18,9 @@
 import uuid
 import json
 from pathlib import Path
+import soundfile
+from os.path import splitext, basename
```
Comment (bastibe) [inline jbof.py]: This should not be necessary. Please use `pathlib` methods instead.

File: setup.py [hunk 4]
```diff
@@ -9,6 +9,6 @@
     url='https://github.com/bastibe/jbof',
     license='GPL v3',
     py_modules=['runforrest'],
-    install_requires=['numpy'],
+    install_requires=['numpy', 'json', 'soundfile', 'scipy'],
```
Comment (bastibe) [inline setup.py]: No need to add `json`, as it is part of the standard lib.

**Review by bastibe — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please change a few minor issues, and then we can merge.

Please also re-add your tests, and update the README with the completed task.

File: setup.py (hunk 4 above)
Comment (zngguvnf) [reply setup.py]: DONE in 4178140

File: jbof.py (hunk 2 above)
Comment (zngguvnf) [reply jbof.py:120]: DONE in 33835db

File: jbof.py (hunk 1 above)
Comment (zngguvnf) [reply jbof.py]: DONE in 9605b75

**Comment by zngguvnf** [comment]
Thanks for the review. This was really helpful for me.
- I fixed the issues you pointed out 
- I updated the README.
- Right now, the test suite is not finished.

**Comment by zngguvnf** [comment]
I added some test, they check:
- can the data be restored.
- Does the metadata include a samplerate, when using an audio format.

**Pull request merged**